Review

OwenSteele opened the pull request.

OwenSteele committed:
Initial commit WIP.

Structure and Repository create, DB create with migration.

OwenSteele committed:
Views done.

OwenSteele committed:
Git Error Fix.

Loss of API and Testing projects - API improved and restored (testing to be done)

OwenSteele committed:
Testing of Web Controller and LocalParks.Data added.

OwenSteele committed:
Fixed Edit Existing ParkEvent View.

Also updated Repo for being made public, removed chached ignored files.

OwenSteele committed:
Create LICENSE

OwenSteele committed:
Update README.md

OwenSteele committed:
Replaced Postcode Enum with Type, 'Remove Tast.Result'

OwenSteele committed:
Added Controller Services.

Removed mulitple responsibilities of Controller by decoupling DB querying and model mapping.

OwenSteele committed:
Created PostcodeModel, updated views.

OwenSteele committed:
Reincluded public shipped API.

Removed two public API controller actions that caused routing ambiguity, fixing an issue which prevented the inclusion of the public API.

OwenSteele committed:
Decoupled API and repo. Add Home.Index elements.

OwenSteele committed:
Added jQuery geolocation for nearest places.

Also made Logo transparents, changed styling of Home.Index.cshtml. fixed lat and long being wrong way round.

OwenSteele committed:
Added test API POST action for parks.

OwenSteele committed:
Updated Home view, model and controller.

OwenSteele committed:
Added additional search filters and sorting to Parks.

Parks can not be sorted by postcode. Any non-collectible type property with a display name can be used to sort the results - solution required for ICollection properties.

OwenSteele committed:
Added sorting and filters to Clubs and Events.

OwenSteele committed:
Fixed parkId queryString not binding.

Added filters/sorting to final controller. Decoupled controller services from reflection.

OwenSteele committed:
Added Token Authentication for web API.

OwenSteele committed:
Interfaced and Added services to config.

OwenSteele committed:
Added contact form and font-awesome CS pkg.

OwenSteele committed:
Added contact form and font-awesome CS pkg. (ignored)

OwenSteele committed:
Added Identity to DB.

Changed from, DbContext to IdentituDbContext. Added partial seeder for user. Added LocalParksUser entity and model. Added AccountController with Login and index views. Alter layout header for login links.

OwenSteele committed:
Completed Jwt Auth Identity on ApiParks POST.

OwenSteele committed:
Added Roles to Identities. Added PUT and DELETE actions to ApiParks.

OwenSteele committed:
Added CRUD to API: Parks, SportsClubs. Supervisor.

OwenSteele committed:
Completed API parkEvents CRUD with Identities.

OwenSteele committed:
Added Account pages, Added Auth to ParkEvents.

OwenSteele committed:
Improved Home/Index view styling.

OwenSteele committed:
Added Creating and Deleting Identities, with UI.

OwenSteele committed:
Added API Postcode EP - See notes.

Added Tokengenerator and ApiDocumentation content. Deprecated function to obtain tokens from public shipped API. Improved Footer design. Fixed issues with Creating and deleting Users.

OwenSteele committed:
Fixed SignUp Success Redirect.

OwenSteele committed:
Added Edit Password and Edit Details functionalilty.

OwenSteele committed:
Moved Account Models to Models.Account.

OwenSteele committed:
Improved Home view - Changed Event lookup to EventId.

OwenSteele committed:
Minor naming fixes.

OwenSteele committed:
Added jQuery client-side validation.

OwenSteele committed:
Improved Closing soon content Home.Index view.

OwenSteele committed:
Removed User Data from source code.

OwenSteele committed:
Minor Home Index View improvements.

OwenSteele committed:
Removed commited data files.

OwenSteele committed:
Data moved to JSON files.

OwenSteele committed:
Seeded Database with JSON instead of Source code.

OwenSteele committed:
Removed commented code.

OwenSteele committed:
Merge pull request #4 from OwenSteele/seedTransition

Seed transition

OwenSteele committed:
Initial commit.

OwenSteele committed:
Added additional seeded Entities.

OwenSteele committed:
Seeded Products and Memberships.

OwenSteele committed:
Added Required labels to views.

OwenSteele committed:
Fixed Deleting Events.

OwenSteele committed:
Merge pull request #5 from OwenSteele/AngularShopDev

Angular shop dev

OwenSteele committed:
Integrated Angular Boilerplate into Project.

OwenSteele committed:
Removed cached files.

OwenSteele committed:
Removed cached files.

OwenSteele committed:
Removed cached files.

OwenSteele committed:
Added Serives and Controller for Shop App.

OwenSteele committed:
Shop typescript boilerplate added, with Issues.

OwenSteele committed:
Added NG Cart, Products and Shop.

OwenSteele committed:
Added checkout to NG shop app.

OwenSteele committed:
Finished angular shop app structure.

OwenSteele committed:
Merge pull request #7 from OwenSteele/AngularShopDev

Angular shop dev

OwenSteele committed:
removed dist files.

OwenSteele committed:
Delete LocalParks/LocalParks/dist/out-tsc directory

OwenSteele committed:
Merge pull request #8 from OwenSteele/AngularShopDev

removed dist files.

OwenSteele committed:
Added angular shop checkout (Issues).

OwenSteele committed:
Added Ordering and View Orders.

OwenSteele committed:
Delete LocalParks/LocalParks/wwwroot/ClientApp directory

OwenSteele committed:
Included grandchildren for user orders and events.

OwenSteele committed:
Merge branch 'master' of https://github.com/OwenSteele/LocalParks

OwenSteele committed:
Added jQuery validation on filters.

OwenSteele committed:
Added viewComponent, ChartJS and Service.

OwenSteele committed:
Added ChartBuilder, created chart.

OwenSteele committed:
ChartBuilder Improvements, added to seeded data.

OwenSteele committed:
Minor View Fixes, added gulp.

OwenSteele committed:
Added loops colors to ChartBuilder.

OwenSteele committed:
Improvements to Home.Index format.

OwenSteele committed:
Added Charts.

OwenSteele committed:
Added Scatter to ChartBuilder, Added additional Charts.

OwenSteele committed:
Changed Report view layout, added additional charts.

OwenSteele committed:
Decoupled Service Concerns.

OwenSteele committed:
Updated Index Views.

OwenSteele committed:
Updated Angular styling.

OwenSteele committed:
Account Domain Dependency Cleanup.

OwenSteele committed:
Shop Domain Dependency Cleanup.

OwenSteele committed:
Home Model and Service Cleanup.

OwenSteele committed:
Added flex-wrap to index views.

OwenSteele committed:
Improved 'Details' and 'Account' Domain Views, See comments.

Fixed Issue with GetUser with decoupling and injecting repo. Improvements to other views.

OwenSteele committed:
Shop checkout button fix, API create token fix.

OwenSteele committed:
Injected ISelectListService in _FilterBar Views, Decoupled from Controllers.

OwenSteele committed:
Updated Testing.

OwenSteele committed:
Initial commit, basic Core tests.

OwenSteele committed:
Added Jumbotron images.

OwenSteele committed:
Moved Configuration to Collections. Updated Views.

OwenSteele committed:
Update README.md

OwenSteele committed:
Update README.md

OwenSteele committed:
Update README.md

OwenSteele committed:
Formatting Fixes.

OwenSteele committed:
Merge branch 'master' of https://github.com/OwenSteele/LocalParks

OwenSteele committed:
Redesigned Home Index View.

OwenSteele committed:
Updated Shop Views, added to Reports Index view.

OwenSteele committed:
Fixed minor issues with editing events. See notes.

Fixed 'date taken' message when editing existing events and no date change. Fixed new events having date of 01/01/0001. Edit Existing events submission now prevented when no changes made with jQuery.

OwenSteele committed:
Removed Deferred Query Execution in Repo.

OwenSteele committed:
Added Repo to Footer.

OwenSteele committed:
Added Repo to Footer.

OwenSteele committed:
Added Query String Encryption. See Notes.

Added Query String encryption middleware, added encryption service (IOptions Pattern did not functions, using Config injection. Service is separate to middleware as removes need to inject a service into middleware. Other minor fixes: Home Page, past events were shown still, Login redirect didn't include parameters.

OwenSteele committed:
Added Remove button to Shop Cart.

OwenSteele committed:
Fixed Park Details Sports Clubs link. Refactored filters code.

OwenSteele committed:
Complete Project refactor. See notes.

Reorientation of project to cleaner architecture style. Moved Business logic from UI App to Infrastructure library. Moved Entities, Contracts and Models to Core and Domain Library. Segregated API into own WebAPI project. STILL TO DO: Segregrate models in the infrastructure with lighter more specific properties, segregate repositories into entity concerns, possibly with CQRS.

OwenSteele committed:
Fixed minor issue with ISelectListService in Event Editing.Updated Shop for production.

OwenSteele committed:
Moved ApiDocs to non-auth domain. Formatted Api Controllers to Swagger Spec.

OwenSteele committed:
Added Swagger to Published API.

OwenSteele committed:
Added new tab targets to external links.

OwenSteele committed:
Added Guest Accounts.

OwenSteele committed:
Update README.md

OwenSteele committed:
Removed Guest Accounts (see notes). Code Cleanup.

OwenSteele committed:
Merge branch 'master' of https://github.com/OwenSteele/LocalParks

OwenSteele committed:
Create dotnet.yml

OwenSteele committed:
Update README.md

OwenSteele committed:
Removed extra folder.

OwenSteele committed:
Merge branch 'master' of https://github.com/OwenSteele/LocalParks

OwenSteele committed:
Update api docs URL. Repo Footer Link opens in new tab. Updated JS event for cookies partial view for hiding on accept.

OwenSteele committed:
Removed 'localparksapiapi' link typo.

OwenSteele committed:
Updated Time Culture. Added Shop Loading message. Changed Api docs margin.

OwenSteele committed:
Merge branch 'master' of https://github.com/OwenSteele/LocalParks

OwenSteele committed:
Merge branch 'master' into review

JamesBaxterSoftware commented on LocalParks.API/Controllers/ApiParkEventsController.cs:
```diff
@@ -0,0 +1,145 @@
+
+                return Ok(results);
+            }
+            catch (Exception ex)
```
You can register exception handler/strategies in your startup class so you don't have to add a try catch in all your controllers apis ... 

These might help.
https://weblog.west-wind.com/posts/2016/oct/16/error-handling-and-exceptionfilter-dependency-injection-for-aspnet-core-apis

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-5.0

OwenSteele replied:
Ah ok, thanks, James. I'll move this and the logging into a handler.

JamesBaxterSoftware commented on LocalParks.API/Controllers/ApiPostcodesController.cs:
```diff
@@ -0,0 +1,68 @@
+namespace LocalParks.API.Controllers
+{
+    [ApiController]
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
```
you can put authorise attribute on base class if all your controllers require authentication

OwenSteele replied:
Ah ok, will do, thanks!

JamesBaxterSoftware commented on LocalParks.API/Controllers/ApiSupervisorsController.cs:
```diff
@@ -0,0 +1,70 @@
+        }
+
+        [HttpGet("api/[controller]/{parkId:int}")]
+        [HttpGet("api/parks/{parkId:int}/[controller]")]
```
two httpget attributes?

OwenSteele replied:
Thought was to have the action accessible from two endpoints for option of how to query it.
Have removed the latter now as it is not used anyway

JamesBaxterSoftware commented on LocalParks.API/Startup.cs:
```diff
@@ -0,0 +1,121 @@
+
+            services.AddCors(options =>
+            {
+                options.AddPolicy("Open", builder => builder.AllowAnyOrigin().AllowAnyHeader().AllowAnyMethod());
```
Be more restrictive here as cor's is designed to stop someone using your api's you can do this in azure instead also up to you.

OwenSteele replied:
Yeah I'll change it to only the Client App endpoint, thanks!

JamesBaxterSoftware commented on LocalParks.Core/Domain/SportType.cs:
```diff
@@ -0,0 +1,28 @@
+﻿namespace LocalParks.Core.Domain
+
+{
+    public enum SportType
```
You could have these in a database table SportType with Columns Id and SportName, then you can add sports without having to do a redeploy of you app everytime.

OwenSteele replied:
That makes sense, thanks, I add that in!

JamesBaxterSoftware commented on LocalParks.Core/Models/Shop/OrderItemModel.cs:
```diff
@@ -0,0 +1,25 @@
+
+        public string ImageId { get; set; }
+
+        public decimal Subtotal { get { return Math.Round(Quantity * UnitPrice, 2); } }
```
When a developer calls a class a Model you would expect there to be no logic when calling get on a property as it is a misleading name. Mayb use a service to get the subtotal whenever needed.

JamesBaxterSoftware commented on LocalParks.Infrastructure.Tests/ChartBuilderTests.cs:
```diff
@@ -0,0 +1,122 @@
+    public class ChartBuilderTests
+    {
+        [Fact]
+        public void WHEN_Chart_Is_Constructed_THEN_Chart_Cannot_be_retrieved()
```
is this chart builder something you found online?

OwenSteele replied:
This is something I wrote as I couldn't find any packages for it, designed to produce charts for the ChartJS package 
https://www.chartjs.org/docs/latest/?q=lazy

JamesBaxterSoftware commented on LocalParks.Infrastructure/Models/ParkProfile.cs:
```diff
@@ -0,0 +1,70 @@
+    {
+        public ParkProfile()
+        {
+            CreateMap<LocalParksUser, LocalParksUserModel>().ReverseMap();
```
automapper is cool and worth having here... but 9/10 it's easier to just write the code if someone asks why you might not want to use it.

OwenSteele committed:
Added ApiExceptionHandler for controllers - Review #14.

Made the MVC and API controllers more lightweight, removed error handling and logging from concerns.

OwenSteele committed:
Changed CORS to restricted, added policies for dev and prod environments - Review #14

OwenSteele committed:
Changed CORS to restricted, added policies for dev and prod environments.

OwenSteele committed:
Merge branch 'review' of https://github.com/OwenSteele/LocalParks into review

OwenSteele committed:
Changed SportType from Enum to Entity and migrated to database with seeded data - Review #14.

OwenSteele committed:
Removed Business Logic from Models, Added InfoServices to replace required actions - Review #14.

OwenSteele committed:
Updated files related to (#14) review comments. See notes.

Injected InfoServices into App Views. Updated the ChartJS Reports Services related to SportType Changes. Added Managers to Infrastructure to take repsonsibilities from, and reduce concerns of MVC controllers, updated controller tests as a result. Moved seededData to ContentRootPath in App.

OwenSteele committed:
Updated AboutMe page.
